### PR TITLE
add a bit about unicode to scaladoc.

### PR DESCRIPTION
**Pull request by jsuereth**
An important fix to the scaladoc for string context.  Review by @retronym

### CODE REVIEW

File: src/library/scala/StringContext.scala [hunk 1]
```diff
@@ -97,6 +97,13 @@ case class StringContext(parts: String*) {
    *
    *  For example, the raw processed string `raw"a\nb"` is equal to the scala string `"a\\nb"`.
    *
+   *  ''Note:'' Even when using the raw interpolator, Scala will preprocess unicode escapes.
+   *  For example: 
+   *  {{{
+   *    scala> raw"\u0123"
+   *    res0: String = ģ
```
Comment (retronym) [inline src/library/scala/StringContext.scala:104]: `\u2603` would have been my choice, but otherwise this LGTM.

Comment (paulp) [inline src/library/scala/StringContext.scala:104]: SNOWMAN BLOCKER

Comment (jsuereth) [inline src/library/scala/StringContext.scala:104]: Dude, if you send me an ASCII scene I will hold up 2.10.0 JUST FOR THAT!

In all seriousness, I can use snowman if it's more "Scala".   If only we there was a unicode unicorn....

Comment (gkossakowski) [inline src/library/scala/StringContext.scala:104]: I vote for snowman too but let's fix it another time.

**Pull request merged**